Index: template.php
===================================================================
RCS file: /cvs/drupal-contrib/contributions/themes/zen/template.php,v
retrieving revision 1.45.2.7
diff -u -p -r1.45.2.7 template.php
--- template.php	26 Jun 2010 19:18:23 -0000	1.45.2.7
+++ template.php	24 Jul 2010 23:28:57 -0000
@@ -228,22 +228,34 @@ function zen_blocks($region, $show_block
     $render_sidebars = $show_blocks;
   }
 
-  // If zen_blocks was called with a NULL region, its likely we were just
-  // setting the $render_sidebars static variable.
+  // Is the Context module enabled? If so make sure that the blocks Context wants to display get displayed 
   if ($region) {
     $output = '';
 
-    // If $renders_sidebars is FALSE, don't render any region whose name begins
-    // with "sidebar_".
-    if (($render_sidebars || (strpos($region, 'sidebar_') !== 0)) && ($list = block_list($region))) {
-      foreach ($list as $key => $block) {
-        // $key == module_delta
-        $output .= theme('block', $block);
+    if (module_exists("context")) {
+      // Get the Context plugin needed to get the blocks that needs to be displayed for this region 
+      $plugin = context_get_plugin('reaction', 'block');
+
+      // Let's get the blocks that should be displayed from Context.
+      if (is_object($plugin)) {
+        $output .= $plugin->execute($region);
       }
     }
+    // If zen_blocks was called with a NULL region, its likely we were just
+    // setting the $render_sidebars static variable.
+    else {
+      // If $renders_sidebars is FALSE, don't render any region whose name begins
+      // with "sidebar_".
+      if (($render_sidebars || (strpos($region, 'sidebar_') !== 0)) && ($list = block_list($region))) {
+        foreach ($list as $key => $block) {
+          // $key == module_delta
+          $output .= theme('block', $block);
+        }
+      }
 
-    // Add any content assigned to this region through drupal_set_content() calls.
-    $output .= drupal_get_content($region);
+      // Add any content assigned to this region through drupal_set_content() calls.
+      $output .= drupal_get_content($region);
+    }
 
     $elements['#children'] = $output;
     $elements['#region'] = $region;
